File: UNIT 02/SOURCE CODE/CSHARP/SeparateFilesExample/MyFunctionsDef.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SeparateFiles_Example
{
    public static class MyFunctionsDef
    {
        public static int factorial(int n)
        {
            return (n == 0) ? 1 : n * factorial(n - 1);
        }

        public  static int MyPower(int x = 1, int n = 0)
        {
            return (n == 0) ? 1 : x * MyPower(x, n - 1);
        }
    }
}

File: UNIT 02/SOURCE CODE/JAVA/SeparateFilesExample/MyFunctionsDef.java
package com.dube.csc213.definition;

/**
 *
 * @author Enock Lukhetfo Dube  <eldube@uniswa.sz>
 */
public class MyFunctionsDef {
    public static int factorial(int n) {
            return (n == 0) ? 1 : n * factorial(n - 1);
    }

    public static int myPower(int x, int n){
            return (n == 0) ? 1 : x * myPower(x, n - 1);
    }
}
